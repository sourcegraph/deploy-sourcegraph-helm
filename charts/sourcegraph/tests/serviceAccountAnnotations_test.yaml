suite: serviceAccountAnnotations
release:
  name: sourcegraph
  namespace: sourcegraph
tests:
- it: should render service account annotations when frontend.serviceAccount.annotations is defined
  set:
    frontend:
      serviceAccount:
        annotations:
          iam.gke.io/gcp-service-account: sourcegraph@sourcegraph.iam.gserviceaccount.com
  template: frontend/sourcegraph-frontend.ServiceAccount.yaml
  asserts:
  - equal:
      path: metadata.annotations
      value:
        iam.gke.io/gcp-service-account: sourcegraph@sourcegraph.iam.gserviceaccount.com

- it: should render service account annotations when cadvisor.serviceAccount.annotations is defined
  set:
    cadvisor:
      serviceAccount:
        annotations:
          iam.gke.io/gcp-service-account: sourcegraph@sourcegraph.iam.gserviceaccount.com
  template: cadvisor/cadvisor.ServiceAccount.yaml
  asserts:
  - equal:
      path: metadata.annotations
      value:
        iam.gke.io/gcp-service-account: sourcegraph@sourcegraph.iam.gserviceaccount.com

- it: should render service account annotations when prometheus.serviceAccount.annotations is defined
  set:
    prometheus:
      serviceAccount:
        annotations:
          iam.gke.io/gcp-service-account: sourcegraph@sourcegraph.iam.gserviceaccount.com
  template: prometheus/prometheus.ServiceAccount.yaml
  asserts:
  - equal:
      path: metadata.annotations
      value:
        iam.gke.io/gcp-service-account: sourcegraph@sourcegraph.iam.gserviceaccount.com
