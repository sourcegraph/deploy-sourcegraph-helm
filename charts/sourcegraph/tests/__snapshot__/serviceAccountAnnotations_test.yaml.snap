should render service account annotations when cadvisor.serviceAccount.annotations is defined:
  1: |
    apiVersion: v1
    kind: ServiceAccount
    metadata:
      annotations:
        iam.gke.io/gcp-service-account: sourcegraph@sourcegraph.iam.gserviceaccount.com
      labels:
        app: cadvisor
        app.kubernetes.io/component: cadvisor
        category: rbac
        deploy: sourcegraph
      name: cadvisor
should render service account annotations when frontend.serviceAccount.annotations is defined:
  1: |
    apiVersion: v1
    kind: ServiceAccount
    metadata:
      annotations:
        iam.gke.io/gcp-service-account: sourcegraph@sourcegraph.iam.gserviceaccount.com
      labels:
        app.kubernetes.io/component: frontend
        category: rbac
        deploy: sourcegraph
      name: sourcegraph-frontend
should render service account annotations when prometheus.serviceAccount.annotations is defined:
  1: |
    apiVersion: v1
    kind: ServiceAccount
    metadata:
      annotations:
        iam.gke.io/gcp-service-account: sourcegraph@sourcegraph.iam.gserviceaccount.com
      labels:
        app.kubernetes.io/component: prometheus
        category: rbac
        deploy: sourcegraph
      name: prometheus
